orchagent/vnetorch.cpp

```diff
@@ -0,0 +1,200 @@
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("Failed to create virtual router name: %s, rv: %d",
+                           vnet_name.c_str(), status);
+            throw std::runtime_error("Failed to create VR object");
+        }
+        return true;
+    };
+
+    /*
+     * Create ingress and egress VRF based on VR_VALID
+     */
+
+    for (auto vr_type : vr_cntxt)
+    {
+        sai_object_id_t router_id;
+        if (vr_type != VR_TYPE::VR_INVALID && l_fn(router_id))
+        {
+            vr_ids.insert(std::pair<VR_TYPE, sai_object_id_t>(vr_type, router_id));
+        }
+    }
+
+    SWSS_LOG_INFO("VNET '%s' router object created ", vnet_name.c_str());
+    return true;
+}
+
+bool VNetVrfObject::updateObj(vector<sai_attribute_t>& attrs)
+{
+    set<sai_object_id_t> vr_ent = getVRids();
+
+    for (const auto& attr: attrs)
+    {
+        for (auto it : vr_ent)
+        {
+            sai_status_t status = sai_virtual_router_api->set_virtual_router_attribute(it, &attr);
+            if (status != SAI_STATUS_SUCCESS)
+            {
+                SWSS_LOG_ERROR("Failed to update virtual router attribute. VNET name: %s, rv: %d",
+                                vnet_name.c_str(), status);
+                return false;
+            }
+        }
+    }
+
+    SWSS_LOG_INFO("VNET '%s' was updated", vnet_name.c_str());
+    return true;
+}
+
+VNetVrfObject::~VNetVrfObject()
+{
+    set<sai_object_id_t> vr_ent = getVRids();
+    for (auto it : vr_ent)
+    {
+        sai_status_t status = sai_virtual_router_api->remove_virtual_router(it);
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("Failed to remove virtual router name: %s, rv:%d",
+                            vnet_name.c_str(), status);
+        }
+    }
+
+    SWSS_LOG_INFO("VNET '%s' deleted ", vnet_name.c_str());
+}
+
+/*
+ * VNet Orch class definitions
+ */
+
+template <class T>
+std::unique_ptr<T> VNetOrch::createObject(const string& vnet_name, set<string>& plist,
+                                          vector<sai_attribute_t>& attrs)
+{
+    std::unique_ptr<T> vnet_obj(new T(vnet_name, plist, attrs));
+    return vnet_obj;
+}
+
+bool VNetOrch::addOperation(const Request& request)
+{
+    SWSS_LOG_ENTER();
+
+    sai_attribute_t attr;
+    vector<sai_attribute_t> attrs;
+    set<string> peer_list = {};
+    bool peer = false, create = false;
+
+    for (const auto& name: request.getAttrFieldNames())
+    {
+        if (name == "src_mac")
+        {
+            const auto& mac = request.getAttrMacAddress("src_mac");
+            attr.id = SAI_VIRTUAL_ROUTER_ATTR_SRC_MAC_ADDRESS;
+            memcpy(attr.value.mac, mac.getMac(), sizeof(sai_mac_t));
+        }
+        else if (name == "peer_list")
+        {
+            peer_list  = request.getAttrSet("peer_list");
+            peer = true;
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Logic error: Unknown attribute: %s", name.c_str());
```

Comment:
>SWSS_LOG_ERROR [](start = 12, length = 14)

you cannot handle this attribute, you still continue, this looks like warning, instead of error.